### PR TITLE
Enabling scenarios tests

### PR DESCRIPTION
**Pull request by fabiocav**

### CODE REVIEW

**Comment by fabiocav** [comment]
@christopheranderson haven't looked at the details but wanted to check before spending time on it. Is the failing test here something you were looking at? Seems to be consistently failing (and looks similar to the failure you were looking at last week): https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-script-y8o14/build/2.0.11606-noqrbtoh/tests

**Comment by christopheranderson** [comment]
We chatted offline, but just adding notes here. Logs are taking too long to get there. Attempt to delay it just lead to other tests failing. Should just mark this one ignored, file a bug, and we'll address it next sprint. Here was my first attempt, but it had side effects. :/ https://github.com/Azure/azure-functions-host/pull/2503

**Comment by fabiocav** [comment]
@christopheranderson disabled the test for now, opened #2511 to track

File: test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs [hunk 1]
```diff
@@ -48,10 +60,11 @@ public ControllerScenarioTestFixture(bool isAuthDisabled)
 
         public HttpClient HttpClient { get; set; }
 
-        public HttpServer HttpServer { get; set; }
+        public TestServer HttpServer { get; set; }
 
-        protected virtual void RegisterDependencies(ContainerBuilder builder, WebHostSettings settings)
+        protected virtual void ConfigureWebHostBuilder(IWebHostBuilder webHostBuilder)
         {
+            webHostBuilder.ConfigureServices(c => c.AddSingleton(HostSettings));
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs]: This hook enables deriving fixtures to register services used by the test server.

File: test/WebJobs.Script.Tests.Integration/Controllers/Keys/KeyManagementFixture.cs [hunk 2]
```diff
@@ -33,9 +38,24 @@ protected override void RegisterDependencies(ContainerBuilder builder, WebHostSe
...
 
-            builder.RegisterInstance<ISecretManager>(SecretManagerMock.Object);
+            webHostBuilder.ConfigureServices(c => c.AddSingleton<ISecretManager>(SecretManagerMock.Object));
 
-            base.RegisterDependencies(builder, settings);
+            base.ConfigureWebHostBuilder(webHostBuilder);
+        }
+
+        public static ApiModel ReadApiModelContent(HttpResponseMessage response)
+        {
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.Integration/Controllers/Keys/KeyManagementFixture.cs]: Added this to workaround a change in behavior in Json.NET when deserializing types inheriting from JObject. Doesn't impact any product code paths.

File: test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs [hunk 3]
```diff
@@ -29,16 +35,22 @@ public ControllerScenarioTestFixture(bool isAuthDisabled)
             HostSettings = new WebHostSettings
             {
                 IsSelfHost = true,
-                ScriptPath = Path.Combine(Environment.CurrentDirectory, @"..\..\..\..\sample"),
+                ScriptPath = Path.Combine(Environment.CurrentDirectory, @"..\..\..\..\..\sample"),
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs]: For the below SecretsPath - where is this path in this repo? There isn't an App_Data folder anymore

Comment (fabiocav) [reply test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs]: Good point! This was not really relevant here, so I was going to use a temp folder missed that. I'll have that updated.

Comment (fabiocav) [reply test/WebJobs.Script.Tests.Integration/Controllers/ControllerScenarioTestFixture.cs]: Updated

**Pull request merged**